src/compiler/program.ts

```diff
@@ -26,6 +27,221 @@ namespace ts {
         return undefined;
     }
 
+    export const defaultInitCompilerOptions: CompilerOptions = {
+        module: ModuleKind.CommonJS,
+        target: ScriptTarget.ES3,
+        noImplicitAny: false,
+        outDir: "built",
+        rootDir: ".",
+        sourceMap: false,
+    }
+
+    export function buildConfigFile(writer: EmitTextWriter, compilerOptions: CompilerOptions, fileNames?: string[], excludes?: string[]) {
+        compilerOptions = extend(compilerOptions, defaultInitCompilerOptions);
+        let { write, writeLine, increaseIndent, decreaseIndent } = writer;
+        let { optionNameMap } = getOptionNameMap();
+        writeConfigFile();
+
+        function writeConfigFile() {

```

Comment:
i do not think we need a full JSON serializer. That was my inital comment. for me this function, without the support for comments, can be written as:
```ts


JSON.stringify(opt, (k, v) => {
    switch (k) {
        case "target": return scriptTargetToString(k);
        case "module": return moduleKindToString(k);
        case "init":
        case "watch":
        case "version":
        case "help":
            return undefined;
        default: return v;
    }
}, "    ");
```